packages/shopify-codemod/test/fixtures/coffeescript-soak-to-condition/assignment.output.js

```diff
@@ -0,0 +1,16 @@
+if (typeof bar === 'function' && bar() != null && bar().baz.qux != null) {
```

Comment:
You're right, I did it this way to try and keep things simple/ without extra intermediary variables. I looked through the admin and it doesn't look like there's anywhere that we do this that would cause issues (that is, no conditionally-called methods that have side effects), it will just be a little less efficient. Not sure if it's best to add the complexity, or to just bail out if a case like this is found.